src/include/migraphx/op/where.hpp

```diff
@@ -42,9 +42,19 @@ struct where
 
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.has(3).same_dims();
+        check_shapes{inputs, *this, true}.has(3).same_dims();
```

Comment:
`same_dims()`. If one input is static and one is dynamic that imply dynamics is fixed because they both need to have `same_dims()`.

Reply:
That would get rejected in Lines  50-54 anyhow.  `if(s1 == s2)...`